R/credentials_gce.R

```diff
@@ -235,7 +335,8 @@ gce_instance_service_accounts <- function() {
 # https://github.com/r-lib/gargle/issues/216
 fetch_gce_access_token <- function(scopes, service_account) {
   path <- glue("computeMetadata/v1/instance/service-accounts/{service_account}/token")
-  response <- gce_metadata_request(path)
+  scope_string <- glue_collapse(scopes, sep = ",")
+  response <- gce_metadata_request(path, query = list(scopes = scope_string))
```

Comment:
This is actually sort of a big change: we now forward the requested scopes to this request for the access token.

This is in response to #216. In my hands, I still can't really see the effect of this, in the sense that my tokens always reflect the instance scopes and not what's requested here. But it also doesn't seem to hurt anything and I can imagine that my super simple GCE usage doesn't cover all of the use cases.